python.d/httpcheck.chart.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# Description: http check netdata python.d module
+# Original Author: ccremer (github.com/ccremer)
+
+import urllib3
+import time
+import re
+
+from bases.FrameworkServices.UrlService import UrlService
+
+try:
+    urllib3.disable_warnings()
+except AttributeError:
+    pass
+
+# default module values (can be overridden per job in `config`)
+# update_every = 2
+priority = 60000
+retries = 60
+
+# Response
+HTTP_RESPONSE_TIME = 'time'
+HTTP_RESPONSE_LENGTH = 'length'
+
+# Status dimensions
+HTTP_SUCCESS = 'success'
+HTTP_BAD_CONTENT = 'bad_content'
+HTTP_BAD_STATUS = 'bad_status'
+HTTP_TIMEOUT = 'timeout'
+HTTP_NO_CONNECTION = 'no_connection'
+
+ORDER = ['response_time', 'response_length', 'status']
+
+CHARTS = {
+    'response_time': {
+        'options': [None, 'HTTP response time', 'ms', 'response', 'httpcheck.responsetime', 'line'],
+        'lines': [
+            [HTTP_RESPONSE_TIME, 'time', 'absolute', 100, 1000]
+        ]},
+    'response_length': {
+        'options': [None, 'HTTP response body length', 'characters', 'response', 'httpcheck.responselength', 'line'],
+        'lines': [
+            [HTTP_RESPONSE_LENGTH, 'length', 'absolute']
+        ]},
+    'status': {
+        'options': [None, 'HTTP status', 'boolean', 'status', 'httpcheck.status', 'line'],
+        'lines': [
+            [HTTP_SUCCESS, 'success', 'absolute'],
+            [HTTP_BAD_CONTENT, 'bad content', 'absolute'],
+            [HTTP_BAD_STATUS, 'bad status', 'absolute'],
+            [HTTP_TIMEOUT, 'timeout', 'absolute'],
+            [HTTP_NO_CONNECTION, 'no connection', 'absolute']
+        ]}
+}
+
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.url = self.configuration.get('url', None)
+        self.regex = re.compile(self.configuration.get('regex', '.*'))
+        self.status_codes_accepted = self.configuration.get('status_accepted', [200])
+        self.follow_redirect = self.configuration.get('redirect', True)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self._manager = None
+
+    def check(self):
+        """
+        Format configuration data and try to connect to server
+        :return: boolean
+        """
+        if not (self.url and isinstance(self.url, str)):
+            self.error('URL is not defined or type is not <str>')
+            return False
+
+        self._manager = self._build_manager()
+        if not self._manager:
+            return False
+
+        self.info('Enabled {url} with (redirect={redirect}, status={accepted}, interval={update}, timeout={timeout}, '
+                  'regex={regex})'
+                  .format(url=self.url, redirect=self.follow_redirect, accepted=self.status_codes_accepted,
+                          update=self.update_every, timeout=self.request_timeout, regex=self.regex.pattern))
+        return True
```

Comment:
I am not sure why we need custom `check() `method here. Please explain.

Reply:
The check method from UrlService makes an initial request before accepting the config. If the service happens to be down when netdata starts the plugin won't activate. Which is, IMO, counterproductive to a httpcheck that is supposed to monitor that very service uptime. 
Since I wanted to be able to use a plugin that can handle service downtime from the beginning, I needed to override. Since the default config does not provide a default `localhost` config, the plugin won't get activated if not configured.

Reply:
default `check()` executes `get_data()`
If `get_data()` returns not empty `dict()`  job will be launched.

default check() should be ok.

Reply:
> before accepting the config.

what do you mean? what is accepting the config?

Reply:
Apparently I was unable to read python code a few days ago. You're right, it's not needed.